### PR TITLE
[1.18] change variable name err to retErr for deferred comparisons

### PR DESCRIPTION
**Pull request by haircommander**
if we're branching on whether we errored in execution at the end of a function, we should use the retErr name, as it should not get shadowed

partial port of https://github.com/cri-o/cri-o/pull/3934/commits

Signed-off-by: Peter Hunt <pehunt@redhat.com>

#### What type of PR is this?

/kind bug

#### Does this PR introduce a user-facing change?

```release-note
fixed bug where pod names would sometimes leak on creation, causing the kubelet to fail to recreate
```

### CODE REVIEW

**Comment by haircommander** [comment]
```
[sig-storage] EmptyDir volumes volume on default medium should have the correct mode [LinuxOnly] [NodeConformance] [Conformance]
```
haven't seen this one before

/retest

File: server/sandbox_run_linux.go [hunk 1]
```diff
@@ -670,13 +670,13 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 	return resp, nil
 }
 
-func setupShm(podSandboxRunDir, mountLabel string) (shmPath string, err error) {
-	shmPath = filepath.Join(podSandboxRunDir, "shm")
+func setupShm(podSandboxRunDir, mountLabel string) (string, error) {
```
Comment (mrunalp) [inline server/sandbox_run_linux.go:673]: Here we are losing documentation implicit in the return value names. If we do this, then we should add a comment for the function.

Comment (mrunalp) [reply server/sandbox_run_linux.go:673]: This case is where I would argue that keeping it as is probably fine as we don't have defers and it is a small function.

Comment (haircommander) [reply server/sandbox_run_linux.go:673]: this one actually snuck...

**Pull request merged**